infrastructure/cdn-in-a-box/varnish/vstats.go

```diff
@@ -0,0 +1,129 @@
+package main
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import (
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+)
+
+type app struct {
+	mu            *sync.Mutex
+	vstats        vstats
+	checkInterval time.Duration
+}
+
+type vstats struct {
```

Comment:
Can this struct be shared with anything that Traffic Monitor already deserializes?

Reply:
It definitely should be shared however it's implemented as just "main" package separated from TC so it's not a package that TM can call. However, I think it should be a package under TC and be built with `pkg` considering statistics will be added too. Should that be in this PR or done separately?

Reply:
If its ok with you, let's keep it separate, since this one is pretty well baked already

Reply:
Yep, will open an issue for it to be done later.